weather.py

```diff
@@ -0,0 +1,440 @@
+# Imports written in alphabetical order
+from os import path
+from csv_reader import CSVReader
+import sys
+from termcolor import colored
+import zipfile
+
+
+class Weatherman:  # Weatherman Application in one class
+
+    file_name_prefix = "Murree_weather_"  # The part of the file name that is same for all files
+
+    # List with months to access by numerical form
+    months = ["Jan", "Feb", "Mar",
+              "Apr", "May", "Jun",
+              "Jul", "Aug", "Sep",
+              "Oct", "Nov", "Dec"
+              ]
+
+    # Class Constructor
+    def __init__(self, path, highest_argument, average_argument, chart_form_argument, chart_form_combined):
+        self.__path = path
+        self.__highest_arguments = highest_argument
+        self.__average_arguments = average_argument
+        self.__chart_form_arguments = chart_form_argument
+        self.__chart_form_combined = chart_form_combined
+        self.files_path_prefix = self.__path + "/weatherfiles/" + Weatherman.file_name_prefix  # path common in all files
+
+        # Report Number to print
+        self.report_num = 1
+
+    def file_exists(self, date):
+
+        if len(date) < 2:  # To check if month is inputted or only year
+            for month in self.months:  # Running code for all 12 months and opening files that exist
+                file = self.files_path_prefix + date[0] + "_" + month + ".txt"
+
+                if path.exists(file):
+                    return True
+
+            raise Exception("Files for this year are not in the system.")
+
+        else:
+            mon = int(date[1])
+            if 0 < mon < 13:  # Validate month is in between 1-12
+                file = self.files_path_prefix + date[0] + "_" + self.months[mon - 1] + ".txt"
+                if path.exists(file):
+                    return True
+            else:
+                raise Exception("Invalid month entered")
+
+            raise Exception("Files for this year are not in the system.")
+
+    def validate_date(self, date):
```

Comment:
Instead of raising an exception just print the error message and return `False` so only current report generation is skipped.
```suggestion
    def validate_year(self, date):
```

Reply:
done